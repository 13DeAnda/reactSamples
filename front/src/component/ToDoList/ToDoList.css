.listItem{
  margin-top: 20px;
  font-size: 20px;
  font-family: cursive;
  list-style: none;
  color: #6a54a5;
}

.deleteButton{
  font-size: 10px;
  margin-top: 10px;
  margin-left: 36px !important;

}

.listContainer{
  margin-left: 10%;
}