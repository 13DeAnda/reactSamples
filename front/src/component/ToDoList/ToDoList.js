import React from 'react';
import '../../shared/shared.css';
import './ToDoList.css';
import _ from 'lodash';

export default class ToDoList extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.addToList = this.addToList.bind(this);
    this.inputSelected = this.inputSelected.bind(this);
    this.deleteSelected = this.deleteSelected.bind(this);
    this.state = {
      textToAdd: "",
      toDoList: [],
      itemsToDelete: 0
    };
  }

  render() {
    return (
      <div className='container text-center'>
        <div className='title'> To Do List </div>
        <div>
          <input type="text" value={this.state.textToAdd} onChange={this.onChange} className='addInput'/>
          <button  onClick={() => this.addToList(this.state.textToAdd)} className='addButton'> Add To List </button>
        </div>
        <ul className='text-left'>
          {this.state.toDoList.map(function(item, index){
            return <li className='listItem' key={index}>
              <input className='checkListBox'
                     checked={item.checked}
                     onClick={() => this.inputSelected(item)}
                     type="checkbox"/>{item.text}
            </li>
          }, this)}
        </ul>
        {this.state.itemsToDelete?
          <button  onClick={this.deleteSelected} className='deleteButton'> Delete Selected </button>
          :null}
      </div>
    );
  }

  deleteSelected() {
    var newList = [];
    for(var i = 0; i < this.state.toDoList.length; i++){
      if(!this.state.toDoList[i].checked){
        newList.push(this.state.toDoList[i]);
      }
    }
    this.setState({toDoList: newList, itemsToDelete: 0});
  }
  inputSelected(item){
    item.checked = !item.checked;
    var itemsToDelete = _.cloneDeep(this.state.itemsToDelete);
    if(item.checked){
      this.setState({itemsToDelete: ++itemsToDelete});
    }
    else{
      this.setState({itemsToDelete: --itemsToDelete});
    }
  }

  addToList (text){
    if(text.length > 0){
     var tempList = this.state.toDoList;
      tempList.push({text:text, checked: false});
      this.setState({toDoList: tempList, textToAdd: ""});
      return tempList;
    }
  }

  onChange (e) {
    this.setState({textToAdd : e.target.value});
  }
}