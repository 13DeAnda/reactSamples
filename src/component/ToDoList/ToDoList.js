import React from 'react';
import '../../shared/shared.css';
import './ToDoList.css';

export default class ToDoList extends React.Component {
  constructor(props) {
    super(props);
    this.onChange = this.onChange.bind(this);
    this.addToList = this.addToList.bind(this);
    this.state = {
      textToAdd: "",
      toDoList: []
    };
  }

  render() {
    return (
      <div className='container text-center'>
        <div className='title'> To Do List </div>
        <div>
          <input type="text" value={this.state.textToAdd} onChange={this.onChange}/>
          <button onClick={() => this.addToList(this.state.textToAdd)}> Add To List </button>
        </div>
        <ul className='text-left'>
          {this.state.toDoList.map(function(item){
            return <li className='listItem' key={item}>
              <input className='checkListBox' type="checkbox"/>{item} 
            </li>
          })}
        </ul>
      </div>
    );
  }

  addToList (text){
    var tempList = this.state.toDoList;
    tempList.push(text);
    this.setState({toDoList: tempList, textToAdd: ""});
    return tempList;
  }

  onChange (e) {
    this.setState({textToAdd : e.target.value});
  }
}