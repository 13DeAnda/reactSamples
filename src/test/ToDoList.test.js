import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';

import ToDoList from '../component/ToDoList/ToDoList';

describe('ToDoList component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ToDoList />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should add to the list array', () => {
    const toDoList = new ToDoList();
    const toDoListArray = toDoList.addToList("someText");
    expect(toDoListArray).toEqual(["someText"]);
  });
});
