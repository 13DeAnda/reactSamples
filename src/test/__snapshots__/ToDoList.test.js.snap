// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList component should render snapshot 1`] = `
<div
  className="container text-center"
>
  <div
    className="title"
  >
     To Do List 
  </div>
  <div>
    <input
      className="addInput"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="addButton"
      onClick={[Function]}
    >
       Add To List 
    </button>
  </div>
  <ul
    className="text-left"
  />
</div>
`;
